test/player_unit.js

```diff
@@ -341,6 +341,21 @@ describe('Player', () => {
         expect(streamingEngine.switchTextStream).not.toHaveBeenCalled();
         expect(streamingEngine.unloadTextStream).not.toHaveBeenCalled();
       });
+
+      it('sets the text stream to null when captions turn off', async () => {
+        await player.setTextTrackVisibility(true);
+        await player.load(fakeManifestUri, 0, fakeMimeType);
+        expect(streamingEngine.switchTextStream).toHaveBeenCalled();
+        expect(shaka.test.Util.invokeSpy(streamingEngine.getCurrentTextStream))
```

Comment:
The invokeSpy wrapper is a compiler hack since the compiler doesn't see jasmine Spy objects as callable.  But in this case, I believe this method is not a Spy, so you should be able to remove the wrapper and call it directly.  Same below.

If I'm wrong, as I often am, build/check.py would complain after removing the wrapper.

Reply:
I just verified, and it seems like there is a Jasmine Spy created for this method in `simple_fakes.js`. Consequently, the compiler complains when I remove the wrapper and call the method directly.

Reply:
Oh, I see.  This is player_unit, so streamingEngine is a fake.  Now the spy part makes more sense.

But since the fix is in StreamingEngine, testing with a fake StreamingEngine doesn't verify the fix.  :-)  You could comment out your one-line fix, and this test would still pass.

You'll need to test this in test/media/streaming_engine_unit.js instead.